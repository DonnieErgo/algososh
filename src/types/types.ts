import { ElementStates } from '../types/element-states';

export interface ICharacter {
  character: string,
  state: ElementStates
}

export interface IArrObject {
  num: number
  state: ElementStates
}