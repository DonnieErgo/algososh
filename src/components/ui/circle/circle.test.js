import renderer from 'react-test-renderer'
import { Circle } from './circle'
import { ElementStates } from '../../../types/element-states'

describe('Circle component tests', () => {

  it('circle renders without letter', () => {
    const circle = renderer.create(<Circle />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with letter', () => {
    const circle = renderer.create(<Circle letter='x' />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with head', () => {
    const circle = renderer.create(<Circle head='x' />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with component in head', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with tail', () => {
    const circle = renderer.create(<Circle tail='x' />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with component in tail', () => {
    const circle = renderer.create(<Circle tail={<Circle />} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with index', () => {
    const circle = renderer.create(<Circle index={1} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with isSmall prop', () => {
    const circle = renderer.create(<Circle isSmall />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with Default state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with Changing state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle renders with Modified state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

})