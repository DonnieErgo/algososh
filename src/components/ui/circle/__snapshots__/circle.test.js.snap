// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests circle renders with Changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with Default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with Modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with component in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with component in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component tests circle renders with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    x
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component tests circle renders with isSmall prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      x
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    x
  </div>
</div>
`;

exports[`Circle component tests circle renders without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
